core/src/test/scala/org/apache/spark/sql/delta/DeltaSourceDeletionVectorsSuite.scala

```diff
@@ -157,15 +158,22 @@ trait DeltaSourceDeletionVectorTests extends StreamTest with DeletionVectorsTest
     val shouldFailAfterCommands = expectations.exists(_.isInstanceOf[ExpectFailure[_]])
 
     val baseActions: Seq[StreamAction] = Seq(
-      StartStream(
-        Trigger.ProcessingTime("10 seconds"),
-        new StreamManualClock(System.currentTimeMillis())),
-      AdvanceManualClock(10L * 1000L),
+      StartStream(Trigger.ProcessingTime(1000), clock),
+      AdvanceManualClock(1000L),
       CheckAnswer((0 until 15): _*),
       AssertOnQuery { q =>
-        // Make sure we only processed a single batch since the initial data load.
```

Comment:
Which check was flaky here?

Reply:
The stream hadn't always stopped processing before we executed the 2 DML commands. So they'd sometimes end up together in the same batch and sometimes in two different batches, because they were processing even before the second increase of the clock. The thing that finally fixed it was adding the waiting for the stream to settle down before even starting on the DML commands.